pkg/runctl/controller_test.go

```diff
@@ -237,31 +238,47 @@ func Test_Controller_syncHandler_mock(t *testing.T) {
 		expectedResult        api.Result
 		expectedState         api.State
 		expectedMessage       string
+		expectedError         error
 	}{
 		{name: "new_ok",
 			pipelineSpec:  api.PipelineSpec{},
 			currentStatus: api.PipelineStatus{},
+			runManagerExpectation: func(rm *runmocks.MockManager, run *runmocks.MockRun) {
+			},
+			expectedResult: api.ResultUndefined,
+			expectedState:  api.StatePreparing,
+		},
+		{name: "preparing_ok",
+			pipelineSpec: api.PipelineSpec{},
+			currentStatus: api.PipelineStatus{
+				State: api.StatePreparing,
+			},
 			runManagerExpectation: func(rm *runmocks.MockManager, run *runmocks.MockRun) {
 				rm.EXPECT().Start(gomock.Any()).Return(nil)
 			},
 			expectedResult: api.ResultUndefined,
 			expectedState:  api.StateWaiting,
 		},
-		{name: "new_fail",
-			pipelineSpec:  api.PipelineSpec{},
-			currentStatus: api.PipelineStatus{},
+		{name: "preparing_fail",
+			pipelineSpec: api.PipelineSpec{},
+			currentStatus: api.PipelineStatus{
+				State: api.StatePreparing,
+			},
 			runManagerExpectation: func(rm *runmocks.MockManager, run *runmocks.MockRun) {
-				rm.EXPECT().Start(gomock.Any()).Return(fmt.Errorf("expected"))
+				rm.EXPECT().Start(gomock.Any()).Return(expectedError)
 			},
-			expectedResult:  api.ResultErrorInfra,
-			expectedState:   api.StateCleaning,
+			expectedResult:  "",
+			expectedState:   api.StatePreparing,
 			expectedMessage: "error syncing resource .*expected",
+			expectedError:   expectedError,
 		},
-		{name: "new_fail_content_error",
+		{name: "preparing_fail_content_error",
```

Comment:
If a content error happened, there should be no error returned by RunManager.Start(). Maybe we need another change here.

Reply:
Let me know when you have time to discuss this.

Reply:
No error on content error anymore.

Reply:
> No error on content error anymore.

In the test case it _does_ return an error.

Reply:
> No error on content error anymore.

What exactly do you mean with this?